### PR TITLE
create a simple lottery

### PR DESCRIPTION
**Pull request by u3IOm4uk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -3,5 +3,12 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery[] arrayLottery = new Lottery[3];
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Application.java:6]: Use one instance of `Lottery`
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
```suggestion
        Lottery lottery = new Lottery();
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import core.basesyntax.ball.Ball;
+import core.basesyntax.ball.Color;
+import java.util.Random;
+
+public class Lottery {
+    public String getRandomBall() {
+        int indexColor = new Random().nextInt(Color.values().length);
+        Color color = Color.values()[indexColor];
+        int number = new Random().nextInt(100);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:11]: All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import core.basesyntax.ball.Ball;
+import core.basesyntax.ball.Color;
+import java.util.Random;
+
+public class Lottery {
+    public String getRandomBall() {
+        int indexColor = new Random().nextInt(Color.values().length);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:9]: `Random` must be a class field
```suggestion
        int indexColor = random.nextInt(Color.values().length);
```

File: src/main/java/core/basesyntax/ball/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.ball;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private int indexColor;
+
+    public String getRandomColor() {
+        indexColor = new Random().nextInt(Color.values().length);
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/ball/ColorSupplier.java:9]: `Random` must be a class field

File: src/main/java/core/basesyntax/Lottery.java (hunk 3 above)
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Lottery.java:9]: Use the `ColorSupplier` to get the random Color
`ColorSupplier` must be a class field

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

File: src/main/java/core/basesyntax/Application.java [hunk 5]
```diff
@@ -3,5 +3,11 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+
+        for (int i = 0; i < 3; i++) {
+            lottery.getRandomBall();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:9]:
```suggestion
```
you are generating ball here, but nobody can see it

File: src/main/java/core/basesyntax/Lottery.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import core.basesyntax.ball.Ball;
+import core.basesyntax.ball.ColorSupplier;
+import java.util.Random;
+
+public class Lottery {
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:8]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    private static final int BOUND = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public String getRandomBall() {
+        int number = random.nextInt(BOUND);
+        return new Ball(colorSupplier.getRandomColor(), number).toString();
+    }
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:17]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+public class Lottery {
+
+    private static final int BOUND = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public String getRandomBall() {
+        int number = random.nextInt(BOUND);
+        return new Ball(colorSupplier.getRandomColor(), number).toString();
+    }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:16]:
```suggestion
    public Ball getRandomBall() {
        int number = random.nextInt(BOUND);
        return new Ball(colorSupplier.getRandomColor(), number);
    }
```

File: src/main/java/core/basesyntax/Application.java [hunk 9]
```diff
@@ -3,5 +3,11 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:5]: remove comments

**Review by okuzan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**